src/util/heap/monotonepageresource.rs

```diff
@@ -261,15 +261,19 @@ impl<VM: VMBinding> MonotonePageResource<VM> {
      }
      }*/
 
-    pub fn reset_cursor(&self, top: Address) {
+    // top might be Address::ZERO
```

Comment:
This is where I think `Option<NonZeroAddress>` can be useful.  See https://github.com/mmtk/mmtk-core/issues/1223